google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkWriter.java

```diff
@@ -1062,7 +1075,11 @@ private void sendOperationLocked(
     bulkCommitBatch.enqueueOperation(op);
     enqueueOperationOnBatchCallback.apply(bulkCommitBatch);
 
-    if (bulkCommitBatch.getMutationsSize() == maxBatchSize) {
+    if (op.getBackoffDuration() > 0) {
```

Comment:
I am a bit confused. Why do we need this and `retryInSmallerBatch`? Is it possible to remove one of these checks?

Reply:
The first check involving `retryInSmallerBatch` ensures that we are adding the retry to an appropriate batch. For example, if the batch already had 15 writes enqueued, and `sendOperationLocked()` is called with a retry, BulkWriter needs to first send the existing batch to create a new batch of size 10.

Consolidated the checks into a single if statement at the start of the function.